src/Stratis.Bitcoin/Base/BaseFeature.cs

```diff
@@ -180,6 +180,7 @@ public override async Task InitializeAsync()
 
             connectionParameters.TemplateBehaviors.Add(new PingPongBehavior());
             connectionParameters.TemplateBehaviors.Add(new ConsensusManagerBehavior(this.chain, this.initialBlockDownloadState, this.consensusManager, this.peerBanning, this.loggerFactory));
+            connectionParameters.TemplateBehaviors.Add(new PHConsensusManagerBehavior(this.chain, this.initialBlockDownloadState, this.consensusManager, this.peerBanning, this.loggerFactory));
```

Comment:
PHConsensusManagerBehavior is for stratis, ConsensusManagerBehavior for non-stratis

Here you have both at the same time, no need to do that.

Reply:
You probably need to use the `Netork..consensus.IsPos` flag, but best would be to use DI.